### ISSUE TITLE
Followers randomly deciding to run away

### ISSUE THREAD

**Issue by logros13** [CleverRaven/Cataclysm-DDA#49941]
### Describe the bug

Npc suddenly ran away from my faction camp and would not follow orders despite still acting like they where in my faction when spoken to.

This is the second time in a few days this has happened on two different worlds. The first time being the one mentioned in the linked issue (which i'm no longer sure is the same bug hence the new report).

### Steps To Reproduce

Ordered the follower to chop wood into planks and went to sleep. After my character woke up i went and talked to my follower selecting not to wake them when i got the message that they where asleep at which point they started wandering away.

### Expected behavior

Followers not randomly deciding to gtfo?

### Versions and configuration

- OS: Windows 10
- Game Version: 89b4763 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    Alternative Map Key [alt_map_key],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    No Rail Stations [No_Rail_Stations]
]

### Additional context

possibly related to #36386? i can adjust their armor though.

traitor follower standing right next to me in this save: 

[Echeta.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6817866/Echeta.zip)

**Comment by logros13** [comment]
Semi-false alarm on the second occurrence? This time they stayed close to the camp (though still running around randomly) and telling them to guard and then follow again seems to have fixed them. There might still be an issue though as the first mentioned follower (which i sadly do not have a save for) took of across the countryside for a good 15 overmap tiles from my camp before getting killed, never tried ordering him to guard so don't know if that would have stopped him.

**Comment by GoLoT** [comment]
For this specific NPC all I can see is that it keeps running around the camp with an activity set, apparently the "tidying up" one. Debugging doesn't show anything wrong with the NPC either. I don't know if that's the expected behavior but I don't think this is related to the linked issue. The NPC still reacts to commands correctly and stays within the camp overmap tile.

**Comment by logros13** [comment]
yeah, will close this until/unless i get a second case like the first npc that took of running across the world.

**Comment by logros13** [comment]
okay, this time there is definitively something screwy going on, has friendly dialogue when spoken to gives a "Errol Madrid is not Friendly!" log message when i try to sort his armor and i cant push/swap position with him,

what happened: sent him out on a camp mission but did not have enough stored calories when he got back. He started to leave but when a talked to him he gave me the "get me an inhaler" mission which i gave him after which he became like this. Note that i never had to re-recruit him after giving him the inhaler, he was immediately acting like he was part of my faction again.

Pretty sure i'm not just being stupid this time @GoLoT 

[Echeta.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6831432/Echeta.zip)
![npcbug](https://user-images.githubusercontent.com/20383070/125965305-c15f49b8-1c95-4aa1-a0ed-8e1efdf000f0.jpg)
![npcbug2](https://user-images.githubusercontent.com/20383070/125965309-2155bccb-8500-4f7c-8c91-4ee1a1b4f4e6.jpg)

**Comment by GoLoT** [comment]
I'll take a look and see if I can figure something out. If he actually quit your faction this time it's possible that one of the following actions left him in an inconsistent state. Probably a bug this time.

**Comment by SariusSkelrets** [comment]
I got NPCs that tried to run away in multiple games

You must ask them to stop their current activity, at least in my case. Mine seem to start wanting to smash non-existent corpses located really far away

**Comment by justn6** [comment]
I've had this happen to me in two different saves. One was on experimental last week, and the latest was on 0xF stable with lots of mods. 

I had 1 follower set to guard a position, and upon returning they had killed something that got in range. When I talk to them and choose the first option "I need you to come with me" the normal follower options come up:
![image](https://user-images.githubusercontent.com/6120105/126912949-ca9f1a1e-4be6-4052-8ae6-1374f4fcd4b7.png)

But the follower does not follow me. If I try to sort their armor, it shows they are not friendly:
![image](https://user-images.githubusercontent.com/6120105/126912923-a1c91cae-903d-4f57-a1e5-c7c88c4339b4.png)

If I enter the player edit screen, it shows they have no faction:
![image](https://user-images.githubusercontent.com/6120105/126913004-0bf335f0-74c1-47a3-acbe-cbc30e2fd7ab.png)

**Comment by tempestnano** [comment]
I came across this bug as well recently. I was able to sidestep it by debugging the NPC to a leader attitude and then talking to them to insist that I lead. I suspect it may be related to debug teleporting.
